### PR TITLE
Unify muted and unmuted migration paths.

### PR DESCRIPTION
**Pull request by boks1971**
If dynacast had disabled all layers, after a migration, the client did not restart publish (it is akin to muted track). That failed migration because migration state machine waits for unmuted tracks to be published (i. e. server has to receive packets).

If a migrating track is in muted state, server does not wait for packets. It synthesises the published event and catches up later when packets actually come in.

Just treating all migrations as the erstwhile muted case. Sythesise publish whether track is muted or not. In the unmuted case, packets might arrive soon after whereas in muted case, it will depend on when unmute happens.

This is tricky stuff. So, will need good testing.

### CODE REVIEW

File: pkg/rtc/participant.go [hunk 1]
```diff
@@ -1572,36 +1587,6 @@ func (p *ParticipantImpl) onSubscribedMaxQualityChange(trackID livekit.TrackID,
 		SubscribedCodecs:    subscribedQualities,
 	}
 
-	// send layer info about max subscription changes to telemetry
```
Comment (boks1971) [inline pkg/rtc/participant.go:1575]: drive-by as I was looking to synthesise this even on migration. Noticed that this is reading a bunch of stuff and synthesising `TrackInfo`. Just got it from track and passing it on as is.

**Review by paulwe — APPROVED** [review]
